resolver/src/lookup.rs

```diff
@@ -62,7 +81,14 @@ impl Lookup {
         rdatas.extend_from_slice(&*self.rdatas);
         rdatas.extend_from_slice(&*other.rdatas);
 
-        Self::new(Arc::new(rdatas))
+        // If both lookups have TTLs, choose the lower one.
+        match (self.valid_until(), other.valid_until()) {
+            (Some(my_ttl), Some(other_ttl)) =>
+                Self::new_with_deadline(Arc::new(rdatas), min(my_ttl, other_ttl)),
+            (Some(ttl), None) | (None, Some(ttl)) =>
+                Self::new_with_deadline(Arc::new(rdatas), ttl),
```

Comment:
It isn't clear that this logic is correct when mixing `Some(ttl)` and `None`. What's the rationale here? Is it even possible that this could happen for `lookup_ip` or is this something that could only happen for other kinds of queries?

OTOH maybe RFC 2308 means we should always have a TTL; if not from the record, then from the overall response? I'm not sure since I don't know all the DNS details.

Regardless, we should write down the reasoning, at least.

Reply:
As far as I can tell, the only time `Lookup::append` is called on a `Lookup` that doesn't have a TTL is here: 
https://github.com/bluejekyll/trust-dns/blob/97afd93b007e0ba4140be4f09ba7245f821f1fa9/resolver/src/hosts.rs#L63-L76

In this case, it seems that the empty lookups are only created as placeholders and should be immediately overwritten with the new lookup, so in this case, the logic seems correct. Everywhere else where `Lookup::append` is called, the lookups are actual responses to a DNS query and should always have a TTL.

I'd be happy to add a comment explaining this in the method, if you like.

Reply:
hosts is an interesting case. It's driven by a file, as opposed to an actual lookup. I wonder if we could remove the optional, and just pick the max_ttl in this case. At the moment, we don't support live reloading of the hosts file. So if this is indeed the only case, then I say we get rid of it by giving it a better default.

Reply:
I believe the other main case where`Lookup`s are constructed without a TTL are the loopbacks. Do you think it's okay for localhost to appear to have MAX_TTL?

Reply:
I just double checked the code, yes, I believe this is safe. My only concern was if this could have some effect on the LRU cache, but I think both of these code paths return before cache is either used. So we don't have to worry about what that would mean.

So yes, I think we can safely simplify this to default to MAX_TTL and thus get rid of the Option on TTL.

Reply:
@bluejekyll okay, sounds good! I'll make that change shortly.

Reply:
@bluejekyll: changed in 526b8d7f4de3bd92ab0ddcbf7d7a2e26895efce8